firebase-firestore/src/main/java/com/google/firebase/firestore/local/LocalDocumentsView.java

```diff
@@ -218,18 +203,21 @@ private MaybeDocument getDocument(DocumentKey key, List<MutationBatch> inBatches
         }
 
         DocumentKey key = mutation.getKey();
-        MaybeDocument baseDoc = results.get(key);
-        MaybeDocument mutatedDoc = mutation.applyToLocalView(baseDoc, batch.getLocalWriteTime());
-        if (mutatedDoc instanceof Document) {
-          results = results.insert(key, (Document) mutatedDoc);
-        } else {
+        MutableDocument document = results.get(key);
+        if (document == null) {
+          document =
+              new MutableDocument(key); // Create invalid document to apply mutations on top of
```

Comment:
ultranit: Consider moving the comment "// Create invalid document..." onto its own line so that the assignment to `document` can also happen on its own line instead of wrapping around.